### DISCUSSION TITLE
What would be the best approach to display the Enum names and not the values

### DISCUSSION THREAD

**Discussion by kiragengis** [allure-framework/allure2#2560] [Questions & Support] [answered]
Hi,

I am using Allure with C# and NUnit. I want to display the Enum name and not the value to improve the readability of the logs. This is a code example:

```
[AllureStep("Processes the payment with {type}")]
public int ProcessInterests(InterestType type)
{
   // Something in here
}

public enum InterestType 
{
    Added,
    Removed
}
```

The actual result in the Logs is something like this: `Processes the payment with '0'`
The expected result in the Logs should be something like this: `Processes the payment with 'Added'`

**Comment by delatrie** [comment, marked as answer]
The right way would be to use the type formatters API:

```csharp
using Allure.Net.Commons;

class ToStringTypeFormatter<T> : TypeFormatter<T>
{
    public string Format(T value) => value?.ToString();
}
```

Then, define a setup fixture in the root namespace (or no namespace at all):

```csharp
using NUnit.Framework;
using Allure.Net.Commons;

[SetUpFixture]
public class MySetup
{
    [OneTimeSetUp]
    public void SetupAllure()
    {
        AllureLifecycle.AddTypeFormatter(new ToStringTypeFormatter<InterestType>());
    }
}
```

Moreover, I believe that should be the default behavior for enums and some other types. I've created an issue and hopefully, will fix that later: https://github.com/allure-framework/allure-csharp/issues/502
